next/test/validation/object.test.ts

```diff
@@ -0,0 +1,25 @@
+import { describe, expect, it } from '@jest/globals'
+import { createHeadlessForm } from '../../src'
+
+describe('object schema validation', () => {
+  it('returns an error if the value is not an object', () => {
```

Comment:
nit: Could we keep the same naming convention as v0 tests. Instead of saying `if`, saying `when` or `given`